subgraphs/enzyme-core/entities/StakeWiseStakingPosition.ts

```diff
@@ -0,0 +1,117 @@
+import { ZERO_BD, logCritical, uniqueEventId } from '@enzymefinance/subgraph-utils';
+import { Address, BigDecimal, BigInt, ethereum, store } from '@graphprotocol/graph-ts';
+import {
+  StakeWiseStakingPosition,
+  StakeWiseStakingPositionChange,
+  Vault,
+  ExternalPositionType,
+  AssetAmount,
+  StakeWiseVaultToken,
+  StakeWiseStakingExitRequest,
+} from '../generated/schema';
+import { getActivityCounter } from './Counter';
+import { useVault } from './Vault';
+
+export function useStakeWiseStakingPosition(id: string): StakeWiseStakingPosition {
+  let position = StakeWiseStakingPosition.load(id);
+  if (position == null) {
+    logCritical('Failed to load stakewise staking position {}.', [id]);
+  }
+
+  return position as StakeWiseStakingPosition;
+}
+
+export function createStakeWiseStakingPosition(
+  externalPositionAddress: Address,
+  vaultAddress: Address,
+  type: ExternalPositionType,
+): StakeWiseStakingPosition {
+  let position = new StakeWiseStakingPosition(externalPositionAddress.toHex());
+  position.vault = useVault(vaultAddress.toHex()).id;
+  position.vaultTokens = new Array<string>();
+  position.active = true;
+  position.type = type.id;
+  position.stakedEthAmount = ZERO_BD;
+  position.save();
+
+  return position;
+}
+
+export function createStakeWiseStakingPositionChange(
+  stakeWiseStakingPositionAddress: Address,
+  changeType: string,
+  stakeWiseVaultToken: StakeWiseVaultToken,
+  assetAmount: AssetAmount,
+  shares: BigDecimal,
+  vault: Vault,
+  event: ethereum.Event,
+): StakeWiseStakingPositionChange {
+  let change = new StakeWiseStakingPositionChange(uniqueEventId(event));
+  change.stakeWiseStakingPositionChangeType = changeType;
+  change.externalPosition = stakeWiseStakingPositionAddress.toHex();
+  change.vault = vault.id;
+  change.stakeWiseVaultToken = stakeWiseVaultToken.id;
+  change.assetAmount = assetAmount.id;
+  change.shares = shares;
+  change.timestamp = event.block.timestamp.toI32();
+  change.activityCounter = getActivityCounter();
+  change.activityCategories = ['Vault'];
+  change.activityType = 'Trade';
+  change.save();
+
+  vault.lastAssetUpdate = event.block.timestamp.toI32();
+  vault.save();
+
+  return change;
+}
+
+export function ensureStakeWiseVaultToken(stakeWiseVault: Address, externalPosition: Address): StakeWiseVaultToken {
+  let id = stakeWiseVault.toHex();
+
+  let stakeWiseVaultToken = StakeWiseVaultToken.load(id);
+  if (!stakeWiseVaultToken) {
+    stakeWiseVaultToken = new StakeWiseVaultToken(id);
+    stakeWiseVaultToken.createdAt = 0;
```

Comment:
As mentioned in the other comment, why are we setting this to 0?

Reply:
Good spot, updated to set this to the block timestamp.